### PR TITLE
feat: :sparkles: `from_documents` in TFiDF retireiver

### PR DESCRIPTION
**Pull request by vempaliakhil96**
# `from_documents` in TFiDF retriever

Hello Langchainers, just added this method in the TFiDF retriever as it was useful and it does not break any code, it can be useful in complex chains to have a simple text-based retriever as one of the tools in your toolkits for text matching docs.

## Before submitting

added `test_from_documents` in `tests/integration_tests/retrievers/test_tfidf.py`

## Who can review?
@dev2049

### CODE REVIEW

**Review by hwchase17 — COMMENTED** [review]
looks great! can we add something to the docs?

**Comment by vempaliakhil96** [comment]
@hwchase17 I did add the docs, but it also shows some unwanted metadata from python notebooks, not sure how to remove it.

File: langchain/retrievers/tfidf.py [hunk 1]
```diff
@@ -34,6 +34,20 @@ def from_texts(
         tfidf_array = vectorizer.fit_transform(texts)
         docs = [Document(page_content=t) for t in texts]
         return cls(vectorizer=vectorizer, docs=docs, tfidf_array=tfidf_array, **kwargs)
+    
+    @classmethod
+    def from_documents(
+        cls,
+        docs: List[Document],
```
Comment (dev2049) [inline langchain/retrievers/tfidf.py:41]: nit: i think most from_documents methods call this arg "documents"

**Review by dev2049 — COMMENTED** [review]
two small comments

File: langchain/retrievers/tfidf.py [hunk 2]
```diff
@@ -34,6 +34,20 @@ def from_texts(
...
+        tfidf_params: Optional[Dict[str, Any]] = None,
+        **kwargs: Any
+    ) -> "TFIDFRetriever":
+        from sklearn.feature_extraction.text import TfidfVectorizer
```
[the remark this answers is not in the document]
Comment (dev2049) [reply langchain/retrievers/tfidf.py:45]: good point, should add `metadatas: Optional[List[dict]]=None` as a kwarg to from_texts and then use that

[the remark this answers is not in the document]
Comment (vempaliakhil96) [reply langchain/retrievers/tfidf.py]: I have updated the logic, now `from_documents` uses `from_texts` and I have added and extra kwarg `metadatas`

**Comment by dev2049** [comment]
made slight changes and landed in #5114. thank you @vempaliakhil96!

**Pull request closed without merging**